### PR TITLE
feat(app-start): Modify screen summary widgets to use query param

### PR DESCRIPTION
**Pull request by narsaynorath**
Removes half the widgets and makes them generally display cold/warm start data depending on the app start type in the selector. Also, since we are now only showing warm or cold start data and not both at the same time, I moved the logic for querying the device class data down into the `DeviceClassBreakdownBarChart` component.

:warning: A new PR will come to update the metrics ribbon in the page header.

After (note, the demo org has no warm start data):

https://github.com/getsentry/sentry/assets/22846452/65767b2d-bb79-4e01-a338-9c1aa58354c8

### CODE REVIEW

File: static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx [hunk 1]
```diff
@@ -9,40 +10,109 @@ import {
...
+  const {
+    data: startupDataByDeviceClass,
+    isLoading,
+    isError,
+  } = useTableQuery({
+    eventView: EventView.fromNewQueryWithPageFilters(
+      {
+        name: '',
+        fields: [
+          startType === 'cold'
```
Comment (narsaynorath) [inline static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx:74]: Most of this logic is the same as when it was in `widgets.tsx`, it's just this field and filters that really change

File: static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx [hunk 2]
```diff
@@ -9,40 +10,109 @@ import {
...
+  const location = useLocation();
+  const {query: locationQuery} = location;
   const {
     primaryRelease,
     secondaryRelease,
     isLoading: isReleasesLoading,
   } = useReleaseSelection();
 
+  const startType = decodeScalar(location.query[SpanMetricsField.APP_START_TYPE]) ?? '';
+  const yAxis = YAXIS_COLUMNS[startType === 'cold' ? YAxis.COLD_START : YAxis.WARM_START];
```
Comment (narsaynorath) [inline static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx:55]: There's a constant I can replace `'cold'` with. I'll update after the first review pass

File: static/app/views/starfish/views/appStartup/screenSummary/startDurationWidget.tsx [hunk 3]
```diff
@@ -105,10 +108,15 @@ function StartDurationWidget({additionalFilters, chartHeight, type}: Props) {
 
   return (
     <MiniChartPanel
-      title={
-        type === COLD_START_TYPE
-          ? t('Avg. Cold Start Duration')
-          : t('Avg. Warm Start Duration')
+      title={startType === 'cold' ? t('Average Cold Start') : t('Average Warm Start')}
```
Comment (edwardgou-sentry) [inline static/app/views/starfish/views/appStartup/screenSummary/startDurationWidget.tsx:111]:
```suggestion
      title={startType === COLD_START_TYPE ? t('Average Cold Start') : t('Average Warm Start')}
```

File: static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx [hunk 4]
```diff
@@ -9,40 +10,109 @@ import {
...
+    yAxes: YAXES,
+    primaryRelease,
+    secondaryRelease,
+  });
+
+  const data = Object.values(
+    transformedData[
+      YAXIS_COLUMNS[startType === 'cold' ? YAxis.COLD_START : YAxis.WARM_START]
+    ]
+  ) as Series[];
```
Comment (edwardgou-sentry) [inline static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx:107]: hmm do we need to type `transformedData` as `{}`?
Feels like you'll get `data` to be `Series[]` without having to do a type assertion if you remove `{}` from `transformedData`

File: static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx [hunk 5]
```diff
@@ -9,40 +10,109 @@ import {
...
+
   return (
     <MiniChartPanel
-      title={title}
+      title={
+        startType === 'cold'
```
Comment (edwardgou-sentry) [inline static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx:112]: also could use the cold constant here

**Review by edwardgou-sentry — APPROVED** [review]
lgtm just a few nits!

File: static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx (hunk 4 above)
Comment (narsaynorath) [reply static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx:107]: Oh you're totally right. I don't remember how that type snuck in there but thanks for catching it!

**Pull request merged**